core/pkg/artifacts/linker.go

```diff
@@ -60,21 +59,21 @@ func (al *ArtifactLinker) Link() error {
 			nil,
 		)
 	default:
-		return fmt.Errorf(
+		err = fmt.Errorf(
 			"LinkArtifact: %s,"+
 				" error: artifact must have either server id or client id",
 			portfolioName,
 		)
 	}
 
 	if err != nil {
-		return fmt.Errorf(
-			"LinkArtifact: %s, error: %v, response: %v",
+		err = fmt.Errorf(
+			"LinkArtifact: %s, error: %w",
 			portfolioName,
 			err,
-			response,
 		)
+		al.Logger.CaptureError(err)
```

Comment:
an error should only be handled once---after you log it, you shouldn't return it.

In this case, it seems better to return the error and let the caller log it.